### PR TITLE
cli: reload RPC-server TLS config on SIGHUP

### LINKED ISSUE #1892: [Feature Request]:Reread configs and TLS certificates on SIGHUP
It would be convinient to update node configuration without full restart.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #1892.

### CODE REVIEW

**Comment by AnnaShaleva** [comment]
I've tested it locally with the help of `kill -1 PID`, do we need a unit-test for that?

File: cli/server/server.go [hunk 1]
```diff
@@ -325,6 +326,21 @@ func startServer(ctx *cli.Context) error {
...
 
+	rpcServerRestartChan := make(chan struct{})
+	sighubCh := make(chan os.Signal, 1)
+	if cfg.ApplicationConfiguration.RPC.TLSConfig.Enabled {
+		signal.Notify(sighubCh, syscall.SIGHUP)
+		go func() {
+			for {
+				switch sig := <-sighubCh; sig {
+				case syscall.SIGHUP:
+					log.Info("SIGHUB received, restarting rpc-server")
```
Comment (roman-khimov) [inline cli/server/server.go:337]: s/SIGHUB/SIGHUP/

**Review by roman-khimov — COMMENTED** [review]
Let's
 * always set SIGHUP handler
 * restart RPC even if TLS is not configured

In future we might also reread whole config, allow some service configuration changes and restart all services.

File: cli/server/server.go [hunk 2]
```diff
@@ -325,6 +326,21 @@ func startServer(ctx *cli.Context) error {
 	go serv.Start(errChan)
 	rpcServer.Start(errChan)
 
+	rpcServerRestartChan := make(chan struct{})
+	sighubCh := make(chan os.Signal, 1)
```
Comment (roman-khimov) [inline cli/server/server.go:330]: s/hub/hup/

**Review by roman-khimov — COMMENTED** [review]
As for the test, we do run node IIRC in our tests, so it should be possible to add a test also.

File: cli/server/server.go (hunk 1 above)
Comment (AnnaShaleva) [reply cli/server/server.go]: Oh-oh, fixed.

**Comment by AnnaShaleva** [comment]
> As for the test, we do run node IIRC in our tests, so it should be possible to add a test also.

We do it via `newTestChain`, but the new piece of code is in the CLI package (not in the rpc/network server). I'm not sure how we can test it via unit-test.

**Comment by roman-khimov** [comment]
Then we can probably leave these tests for #1576.

File: cli/server/server.go [hunk 3]
```diff
@@ -337,7 +351,17 @@ Main:
 			shutdownErr = fmt.Errorf("server error: %w", err)
 			cancel()
 
+		case <-rpcServerRestartChan:
```
Comment (roman-khimov) [inline cli/server/server.go:354]: Why can't you listen `sighupCh` here directly?

Comment (AnnaShaleva) [reply cli/server/server.go]: The original design required a separate channel, but it's OK now to use `sighupCh`, so fixed.

File: cli/server/server.go [hunk 4]
```diff
@@ -336,8 +340,20 @@ Main:
...
+		case sig := <-sighupCh:
+			switch sig {
+			case syscall.SIGHUP:
+				log.Info("SIGHUP received, restarting rpc-server")
+				serverErr := rpcServer.Shutdown()
+				if serverErr != nil {
+					errChan <- fmt.Errorf("error while restarting rpc-server: %w", serverErr)
+					break
+				}
+				rpcServer = server.New(chain, cfg.ApplicationConfiguration.RPC, serv, serv.GetOracle(), log)
```
Comment (fyrchik) [inline cli/server/server.go:352]: This way config won't be reread.

Comment (AnnaShaleva) [reply cli/server/server.go:352]: Yes, that's exactly the way we'd like it to work. IIUC, we don't want to reload the config, we want to re-read the cert files with old names (the files may be updated by this moment).

**Review by roman-khimov — APPROVED** [review]

**Comment by roman-khimov** [comment]
One thing that probably can be improved is documentation, how will anyone find out anything about this feature?

**Pull request merged**